### PR TITLE
RSDK-3381 - Add defaults to streamStatus

### PR DESCRIPTION
**Pull request by cheukt**
came across while writing tests

### CODE REVIEW

File: src/robot/client.ts [hunk 1]
```diff
@@ -635,13 +635,13 @@ export class RobotClient implements Robot {
...
-    duration: Duration
+    resourceNames?: ResourceName[],
+    duration?: Duration
   ): RobotStatusStream {
     const { robotService } = this;
     const request = new proto.StreamStatusRequest();
-    request.setResourceNamesList(resourceNames);
-    request.setEvery(duration);
+    request.setResourceNamesList(resourceNames ?? []);
+    request.setEvery(duration ?? new Duration().setNanos(500_000_000));
```
Comment (maximpertsov) [inline src/robot/client.ts:644]:
```suggestion
    request.setEvery(new Duration().setNanos(durationMs * 1e+6));
```

File: src/robot/robot.ts [hunk 2]
```diff
@@ -138,12 +138,13 @@ export interface Robot {
...
+   *   Default is [].
+   * @param duration - How often to send a new status. Default is 0.5 seconds.
    * @group Status
    * @alpha
    */
   streamStatus(
-    resourceNames: ResourceName[],
-    duration: Duration
+    resourceNames?: ResourceName[],
+    duration?: Duration
```
Comment (maximpertsov) [inline src/robot/robot.ts:148]:
```suggestion
    durationMs?: number
```

**Review by maximpertsov — CHANGES_REQUESTED** [review]
good catch - just a minor suggestion for the api

**Comment by cheukt** [comment]
merge suggestions and added fixes - tested with
```
  const names = await robot.resourceNames();
  console.log(names[0]);
  console.log(await robot.getStatus([names[0]]));
  let stream = robot.streamStatus();
  stream.on('data', (response: VIAM.robotApi.Status[]) => {
    console.log(response);
  });
  stream.on('status', (newStatus?: { details: unknown }) => {
    console.error('error streaming robot status', newStatus);
  });
  stream.on('end', () => {
    console.error('done streaming robot status');
  });
```
+ variations

**Review by maximpertsov — APPROVED** [review]

**Pull request merged**